### DISCUSSION TITLE
tenant/workspace/namespace isolation

### DISCUSSION THREAD

**Discussion by wwulfric** [vanna-ai/vanna#715] [Ideas]
It would be perfect to have a tenant/workspace/namespace or other kind of concept to isolate ddl and question_sql trainning data. When calling `vanna.ask`, we can only retrieve data from a specific group of training data

**Comment by avinashkurup** [comment]
Yes definitely, @wwulfric ! 

Hi @zainhoda,

This feature would be a game-changer for production use! The ability to isolate DDL and question-SQL training data within a tenant/workspace/namespace would make vanna.ask much more viable for multi-tenant environments.

Is this something on the roadmap? Would love to hear your thoughts.
